app/components/App.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import { Route } from 'react-router-dom';
+import Axios from 'axios';
+import PropTypes from 'react-proptypes';
+import MovieContainer from './MovieContainer';
+import MovieTrailerPlayer from './MovieTrailerPlayer';
+
+class App extends Component {
+  state = {
+    movies: [],
+  };
+
+  componentDidMount() {
+    const { config } = this.props;
+    Axios.get(config.API_URL).then(({ data = [] }) => {
+      this.setState({ movies: data });
+    });
+  }
+
+  render() {
+    const { movies } = this.state;
+    const { config } = this.props;
+
+    return (
+      <div>
+        <Route exact path="/" render={() => <MovieContainer movies={movies} />} />
```

Comment:
Be careful when you use arrow functions. Here there could be a performance issue because MovieContainer are mounted each time that render runs

> “If you use arrow functions within render, each call to render will create new function objects.”

https://frontarm.com/james-k-nelson/when-to-use-arrow-functions/

Reply:
@FBieg just read the article and I'm conscious about this issue. I would like instead of using `render` property of the route (which currently uses the arrow function) to use the `component` property so we can send it directly the component we want to route to. However, in this specific case I need to send the movies array to the MovieContainer component so it can display the movies and since I need it to send it as props for this moment I need to use the `render` route property and the arrow function. Here's what I propose, since movies is a kind of global data that could be needed for more components I plan to implement either Redux or Context API for global state management so I can add movies as a global state and with this we avoid the props cascading issue through the projects code. I'm not implementing global state management shortly since I need to practice testing in this project and styled components with Emotion and I need to build an UI for this project and I need this PR to be merged. So what I will do is to create a GitHub issue/ticket for this project with your comment/concern about this arrow function potential performance issue so after I implement the UI and the tests then I'll start working with global state management (Either Redux or Context API) so we can fix this issue and refactor everything on top of that implementation. What do you think ?

Reply:
@FBieg here's the issue created: https://github.com/jpolanco91/react-movie-app/issues/3, if you agree with this I'll push the other changes to address feedbacks so I can merge the PR, then implement the UI with Emotion, then the tests and after practicing that, I'll take the issue and use either Redux or Context and do a complete code refactor so we can remove that potential performance issue.

Reply:
@jpolanco91 Since these components are too smaller, that issue will not affect the performance in a way that you will notice it. However, if in the future you work on a component with several children, just remember this issue.

In this particular case, we can leave it just like that 👍 

PD: You're doing a great job until now 😃

Reply:
Thanks @FBieg